### DISCUSSION TITLE
CloudSmith mirrors missing aarch64 content

### DISCUSSION THREAD

**Discussion by mnasiadka** [rabbitmq/rabbitmq-server#9553] [Other] [answered]
Hello,

Since CloudSmith directly is a no go right now (see [1]) - would it be possible to mirror the aarch64 content also on the mirrors?

[1]: https://github.com/rabbitmq/rabbitmq-server/discussions/9526

**Comment by michaelklishin** [comment]
@mnasiadka there were never aarch64 versions of the RabbitMQ package on either Cloudsmith or PackageCloud. 

RabbitMQ packages are entirely architecture-neutral.
As for Erlang, [our zero dependency Erlang RPM](https://github.com/rabbitmq/erlang-rpm/) provides aarch64 packages as direct downloads.

Our team does have access to aarch64 workers for Concourse. This is open source software, so anyone
who needs aarch64 versions can host their own packages. Our small team does not have the resources to produce and host every package type in existence.

**Comment by michaelklishin** [comment]
@mnasiadka the OpenStack community keeps coming to us with package type requests. Instead of repeatedly doing so, consider hosting their own mirror (however you please).

Some Fedora engineers publish our zero dependency Erlang RPM to Fedora COPR. I don't see why the same approach would not work for an architecture-neutral package like RabbitMQ itself.

**Comment by michaelklishin** [comment]
Our mirrors use

```
reposync -y --gpgcheck --download-metadata --repoid "modern-erlang-el9" -p /local/path

reposync -y --gpgcheck --download-metadata --repoid "rabbitmq-el9" -p /local/path
```

where both upstream repos are Cloudsmith. I suspect that `reposync` uses the local architecture
to determine what it mirrors.

If that hypothesis is correct, it would take x2 the number of mirrors (two more VMs right now)
to host aarch64 packages. And DigitalOcean does not seem to provide aarch64 packages to begin with.

So Fedora COPR seems to be the way to go for those who are interested in solving this (as opposed to asking Team RabbitMQ to provide everything for everyone).

**Comment by michaelklishin** [comment, marked as answer]
Someone tells me members of the OpenStack community have previously told us that we "have removed" aarch64 packages from Cloudsmith. So there must be an urban myth that arch-specific RabbitMQ packages were once available and were removed at some point.

We've had this conversation before several times and do not have anything new to add.

 * aarch64 Erlang RPMs packages are available [from GitHub](https://github.com/rabbitmq/erlang-rpm/releases) and there's just one file to download
 * aarch64 Debian packages of Erlang are [available from Launchpad](https://launchpad.net/~rabbitmq), starting with Erlang 24 and all the way to the latest 26.1. There is a downside: Launchpad by design keeps only the most recent package version, most likely for obvious cost reasons
 * RabbitMQ packages are all architecture-neutral and [entirely open source](https://github.com/rabbitmq/rabbitmq-packaging)

The OpenStack community can either use services such as Fedora COPR and Launchpad, or host their own mirror, or pay for their own Cloudsmith, PackageCloud, or any similar service, and upload OSS RabbitMQ and our zero dependency Erlang packages there.
